### ISSUE TITLE
Error when step size is larger than maximum position or rows in table

### ISSUE THREAD

**Issue by xiekunwhy** [tavareshugo/WindowScanR#4]
Hi,
I want to calculate some statistics of non-overlap windows, but when I set step size equal to window size, the function gave ERRORs and stoped.
 pos_win <- winScan(x = is_table, 
                    groups = "Chr", 
                    position = "Pos", 
                    values = c("C_m_1", "EL"), 
                    win_size = 500,
                    win_step = 500,
                    funs = "mean")
Error in names(out) <- `*vtmp*` : 
  'names' attribute [4] must be the same length as the vector [3]
In addition: Warning message:
In names(out)[4:ncol(out)] <- as.vector(t(outer(values, c("n", funs),  :
  number of items to replace is not a multiple of replacement length

**Comment by tavareshugo** [comment]
Dear @xiekunwhy,

Unfortunately, I could not reproduce your error. If I do the following, there's no problem:

```
library(dplyr)
library(windowscanr)

# Make up some raw data
raw_data <- data.frame(chr = rep(c("Chr1", "Chr2"), each = 400), 
                       pos = c(sort(sample(1:600, 400)), sort(sample(1:1000, 400))),
                       value1 = rnorm(800), 
                       value2 = rnorm(800))

# Run the window summary
window_summary <- winScan(raw_data,
        groups = "chr",
        position = "pos",
        values = c("value1", "value2"),
        win_size = 50,
        win_step = 100,
        funs = "mean")
```

Could you please provide with a [reproducible example](https://stackoverflow.com/questions/5963269/how-to-make-a-great-r-reproducible-example) for your case?

**Comment by xiekunwhy** [comment]
Sorry, these error  just appear when total length of "pos" is smaller than window size. Max pos of groupA is 1000, max pos of groupB is 50000000, when I use window size 10000, the error appeared!
Line 80th in method-winScan.R (win_start <- win_start[-length(win_start)]) cause the error, so I think you need to consider the situation I was met above.
[test10000.score.txt](https://github.com/tavareshugo/WindowScanR/files/2188187/test10000.score.txt)
here is the example data I was used.

```
is <- fread("test10000.score.txt", header = T)
CountNA <- function(x){sum(is.na(x))}
sum1 <- function(x){sum(x, na.rm = T)}
pos_win_na <- winScan(x = is, 
                   groups = "Chr", 
                   position = "Pos",
                   values = names(is)[3:ncol(is)], 
                   win_size = 50000,
                   win_step = 50000,
                   funs = c("CountNA", "sum1"))
write.table(pos_win_na, file = "allele_counts.txt", col.names = T, row.names = F, quote = F, sep = "\t")
```

**Comment by tavareshugo** [comment]
@xiekunwhy 

OK, I got what the problem was, thanks for spotting it.

I think it's fixed now, you should re-install the package with the command (better to start a new R session first):

`devtools::install_github('tavareshugo/windowscanr')`

Note that I've also fixed #2, so window start doesn't start at zero anymore either.

I've closed the issue but let me know if there's still a problem.
